pages/_app.js

```diff
@@ -149,34 +201,22 @@ export default function App({ Component, pageProps }) {
       setIsDuplicateError(true);
       return;
     }
-
-    setCategories(
-      categories.map((category) =>
-        category.id === editedCategory.id ? editedCategory : category
-      )
-    );
-    setSuccessMessage("Category successfully updated!");
-    setIsEditCategory(null);
   }
 
   function handleDeleteCategory(category) {
-    setCategoryToDelete(category.name);
+    setCategoryToDelete(category);
```

Comment:
typically this would be done via just using the "_id" to identify the respective category that needs to be deleted. but as long this works its ok.